### PR TITLE
add RPN solver

### PR DESCRIPTION
**Pull request by kriadmin**
## What does your PR belong to?
- [ ] Website
- [x] Snippets
- [ ] General / Things regarding the repository (like CI Integration)

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking improvement of a snippet)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have checked that the changes are working properly
- [x] I have checked that there isn't any PR doing the same
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: snippets/RPNSolver.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+Solves the given reverse polish notation
+```js
+const RPNSolver = RPN => {
+  const isNumeric = str => !isNaN(parseFloat(str)) && isFinite(str);
+  const isOperator = str => ['*','-','+','/','**'].includes(str)
+  let stack = [];
+  let solve  = RPN.replace(/\^/g,'**').split(/\s+/g);
+  solve.forEach(symbol => {
+    if(isNumeric(symbol)) {stack.push(symbol)}
```
Comment (Chalarangelo) [inline snippets/RPNSolver.md]: You could inline the check for numeric value in the parenthesis to save a line.

File: snippets/RPNSolver.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+          else if (isOperator(symbol)){
```
Comment (Chalarangelo) [inline snippets/RPNSolver.md]: You could definitely save a line by inlining the check in this statement.

File: snippets/RPNSolver.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+          a = stack.pop();
+          b = stack.pop();
+          if(symbol === "+") {
```
Comment (Chalarangelo) [inline snippets/RPNSolver.md]: This `if... else if...` chain can be written as a switch or even using some `Array.prototype` method (maybe a bit too complex in that case).

File: snippets/RPNSolver.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+const RPNSolver = RPN => {
+  const isNumeric = str => !isNaN(parseFloat(str)) && isFinite(str);
+  const isOperator = str => ['*','-','+','/','**'].includes(str)
+  let stack = [];
+  let solve  = RPN.replace(/\^/g,'**').split(/\s+/g);
+  solve.forEach(symbol => {
+    if(isNumeric(symbol)) {stack.push(symbol)}
+          else if (isOperator(symbol)){
+          a = stack.pop();
+          b = stack.pop();
```
Comment (Chalarangelo) [inline snippets/RPNSolver.md]: These two statements could be `[a,b] = [stack.pop(), stack.pop()]` in order to save a line (provided that works and I am not missing something).

File: snippets/RPNSolver.md (hunk 2 above)
Comment (kriadmin) [reply snippets/RPNSolver.md]: Inlining means?

**Comment by Chalarangelo** [comment]
For mapping symbols to operations you can do the following: 
```js
const operators = {
 '*' : (a,b) => a * b,
 '+' : (a,b) => a + b,
 '-' : (a,b) => a - b,
 '/' : (a,b) => a / b,
 '**': (a,b) => a ** b
}
```

Then, you can check if something is an operator using `Object.keys(operators).includes(operator)`.

And you can do operations using `stack.push(operators[operator](parseFloat(b),parseFloat(a)));`.

File: snippets/RPNSolver.md (hunk 2 above)
Comment (Chalarangelo) [reply snippets/RPNSolver.md]: `else if (['*','-','+','/','**'].includes(symbol))`

**Comment by kriadmin** [comment]
@Chalarangelo Yes Please

File: snippets/solveRPN.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  solve.forEach(symbol => {
+    if(!isNaN(parseFloat(symbol)) && isFinite(symbol)) {stack.push(symbol)}
+          else if (Object.keys(OPERATORS).includes(symbol)) {
+          const [a,b] = [stack.pop(),stack.pop()];
+          stack.push(OPERATORS[symbol](parseFloat(b),parseFloat(a)))
+        } else { throw `${symbol} is not a recognized symbol` }
+  }
+)
+ if(stack.length === 1)  return stack.pop()
+  else  throw `${rpn} is not a proper RPN. Please check it and try again`
```
Comment (skatcat31) [inline snippets/solveRPN.md]: The else here is kind of uneeded since a return short circuits. Throw also short circuits

**Pull request merged**